src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -186,11 +186,11 @@ void MutableObjectManager::DestroySemaphores(const ObjectID &object_id) {
   // semaphores below. As the two semaphores have already been unlinked by the first
   // instance, the sem_unlink() calls below will both fail with ENOENT.
   int ret = sem_unlink(GetSemaphoreHeaderName(name).c_str());
-  if (ret) {
+  if (ret != 0) {
```

Comment:
I feel like we should have macros to handle syscall errors.

Reply:
added todo comment for follow-up, wanna keep changes minimal to clang-tidy stuff here, ruiyang's comment mentioned trying to keep this limited for review purposes

Reply:
a nit on comment, I really prefer to have a macro for syscall, since I think we heavily do syscall in our codebase.